test/packages/EABase/include/Common/EABase/eahave.h

```diff
@@ -559,7 +559,7 @@
 #endif
 
 #if !defined(EA_HAVE_ISNAN) && !defined(EA_NO_HAVE_ISNAN)
-	#if defined(EA_PLATFORM_MICROSOFT) && !defined(EA_COMPILER_CLANG)
+	#if defined(EA_PLATFORM_MICROSOFT) && !defined(EA_PLATFORM_MINGW)
```

Comment:
https://github.com/electronicarts/EASTL/blob/master/test/packages/EABase/include/Common/EABase/eahave.h#L562

Since both Mingw's clang and Mingw's gcc define EA_PLATFORM_MICROSOFT, but don't have the _isnan and _finite symbols, we should remove mingw in its entirety from the expression.
The reason I added EA_COMPILER_CLANG earlier on that line is because I didn't test for MingW's gcc at the time.

Since MingW exposes isnan and isinf, we can remove win clang from the apple expression, since std::isnan is just `namespace std { using ::isnan; }` anyway.

I'm pretty sure that would work for clang-cl since that uses the msvc libraries.

Thus I still think the expression how it is now is correct, but that we should roll back the apple expression to being purely apple again.